contracts/validator-manager/tests/PoSValidatorManagerTests.t.sol

```diff
@@ -882,104 +691,189 @@ abstract contract PoSValidatorManagerTest is ValidatorManagerTest {
             startTime: completeRegistrationTimestamp
         });
         posValidatorManager.completeDelegatorRegistration(0, delegationID);
+    }
+
+    function _registerDefaultDelegator(bytes32 validationID)
+        internal
+        returns (bytes32 delegationID)
+    {
+        return _registerDelegator({
+            validationID: validationID,
+            delegatorAddress: DEFAULT_DELEGATOR_ADDRESS,
+            weight: DEFAULT_DELEGATOR_WEIGHT,
+            initRegistrationTimestamp: DEFAULT_DELEGATOR_INIT_REGISTRATION_TIMESTAMP,
+            completeRegistrationTimestamp: DEFAULT_DELEGATOR_COMPLETE_REGISTRATION_TIMESTAMP,
+            expectedValidatorWeight: DEFAULT_DELEGATOR_WEIGHT + DEFAULT_WEIGHT,
+            expectedNonce: 1
+        });
+    }
+
+    function _registerDelegator(
+        bytes32 validationID,
+        address delegatorAddress,
+        uint64 weight,
+        uint64 initRegistrationTimestamp,
+        uint64 completeRegistrationTimestamp,
+        uint64 expectedValidatorWeight,
+        uint64 expectedNonce
+    ) internal returns (bytes32) {
+        bytes32 delegationID = _setUpInitializeDelegatorRegistration({
+            validationID: validationID,
+            delegatorAddress: delegatorAddress,
+            weight: weight,
+            registrationTimestamp: initRegistrationTimestamp,
+            expectedValidatorWeight: expectedValidatorWeight,
+            expectedNonce: expectedNonce
+        });
+        _setUpCompleteDelegatorRegistration(
+            validationID,
+            delegationID,
+            completeRegistrationTimestamp,
+            expectedValidatorWeight,
+            expectedNonce
+        );
         return delegationID;
     }
 
-    function _setUpInitializeEndDelegation(
+    function _initializeEndDelegationValidatorActiveWithChecks(
```

Comment:
Is there a specific use case you have in mind that would be better served by having separate `WithChecks` functions? It looks like the functions without the checks are not called directly anywhere else.

Reply:
If you expect the code to revert somewhere inside of `_initializeEndDelegationValidatorActive`, you can call `vm.expectRevert` before calling it. If you used `_initializeEndDelegationValidatorActiveWithChecks` in the same situation, your test would probably fail because it failed to emit an event that it was expecting, due to the early revert.